components/FrontMatter.js

```diff
@@ -0,0 +1,27 @@
+import { Text } from "@asyncapi/generator-react-sdk";
+
+import * as fs from "fs";
+import * as yaml from "yaml";
+
+export function FrontMatter({ asyncapi, params }) {
+
+  if (!(params.slate || params.frontMatter)) return '';
```

Comment:
In description of `frontMatter` parameter we read `The name of a JSON or YAML formatted file containing values to override the default YAML frontmatter.` and this logic denies it. If someone pass `frontMatter` parameter then we should only read content and replace default content for slate, so we should change it to

```suggestion
  if (!params.slate) return '';
```

Reply:
I think I need to reword the parameter description instead. The `frontMatter` parameter can be used even when the `slate` parameter is false.

Reply:
Also if `slate` and `frontMatter` must be true, then default slate content, I mean  `frontMatter = { title: asyncapi.info().title(), ... }` is unnecessary.

Reply:
Excuse me... My mistake. The logic is correct.

Reply:
They are independent parameters.

Reply:
Yes, independent, but as I understand, if `slate` is set to `true` then we disable rendering of markdown TOC and render slate, so `FrontMatter` should only be rendered if `slate` is true, even if `frontMatter` is set or not, so I think that `if (!params.slate) return '';` is correct. Checking `frontMatter` parameter is unnecessary in first condition.

Reply:
No, if parameter `frontMatter` is set, the contents should be read and rendered, regardless of the `slate` parameter. The ToC would then be unaffected.

Reply:
Ok, it explains a lot. 👍🏼 I would like to ask you to remove the unnecessary `if` in the `TableOfContent` component. https://github.com/asyncapi/markdown-template/pull/71#discussion_r581935577 For me everything will be correct.
@derberg do you want to check it yourself